### PR TITLE
ci: More unsafe directory fixes

### PR DESCRIPTION
**Pull request by javierhonduco**
To fix the broken CI (https://github.com/parca-dev/parca-agent/pull/1287)

Signed-off-by: Francisco Javier Honduvilla Coto <javierhonduco@gmail.com>

### CODE REVIEW

**Review by kakkoyun — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -53,6 +53,9 @@ jobs:
       - name: Check out the code
         uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
 
+      - name: Add repository directory to the git global config as a safe directory
+        run: git config --global --add safe.directory /__w/parca-agent/parca-agent
```
Comment (maxbrunet) [inline .github/workflows/build.yml:57]: This one should not be necessary, I've only seen affecting job containers so far

**Review by maxbrunet — COMMENTED** [review]
Thanks for doing that, as Git get upgraded in the container images we see more of these, I'm going to add a few comments for the reason

File: .github/workflows/build.yml (hunk 1 above)
Comment (maxbrunet) [reply .github/workflows/build.yml]: The snap workflow can use one tho

Comment (javierhonduco) [reply .github/workflows/build.yml]: Thanks!!

**Pull request merged**